### PR TITLE
WT-10927 Re-enable hs verification

### PR DESCRIPTION
**Pull request by clarissecheah**
Re-enable HS verification code that was disabled as part of [WT-10925](https://jira.mongodb.org/browse/WT-10925). Additionally, fix a small bug in HS verification code that was malloc-ing in a loop, but only free-ing once upon exiting the loop.

### CODE REVIEW

File: test/suite/test_bug030.py [hunk 1]
```diff
@@ -122,4 +122,4 @@ def test_bug030(self):
             evict_cursor.search()
         evict_cursor.reset()
         evict_cursor.close()
-        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(70))
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(70))
```
Comment (kommiharibabu) [inline test/suite/test_bug030.py:125]: This change is unnecessary, revert it.

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: test/suite/test_bug030.py (hunk 1 above)
Comment (etienneptl) [reply test/suite/test_bug030.py:125]: Yes, missing last line.

**Review by etienneptl — APPROVED** [review]

File: test/suite/test_bug030.py (hunk 1 above)
Comment (clarissecheah) [reply test/suite/test_bug030.py]: Fixed in [d18f392](https://github.com/wiredtiger/wiredtiger/pull/9363/commits/d18f3929bb049cf5a9287d5ca78224379f0aaa3d)

**Review by jiechenbo — APPROVED** [review]
LGTM!

**Comment by clarissecheah** [comment]
evergreen merge

**Pull request merged**